packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -426,6 +430,15 @@ export class StateMachine extends StateMachineBase {
     this.role.addToPrincipalPolicy(statement);
   }
 
+  private validateStateMachineName(stateMachineName: string) {
+    if (stateMachineName.length < 1 || stateMachineName.length > 80) {
+      throw new Error('StateMachine name length must be between 1 and 80 characters');
+    }
+    if (stateMachineName.match('[0-9a-zA-Z\_\-]+')?.length != 1) {
```

Comment:
can this be simplified to just use a regex with the match.

```suggestion
    if (stateMachineName.match('/^[0-9a-zA-Z\_\-]+$/')) {
```

not 100% sure, but I don't think the `.length` bit would be needed. There's a few examples within the repo.

Reply:
Yeah here's a similar example to what you suggest: https://github.com/aws/aws-cdk/blob/404b5569bb76e3f579dd3005dcf06eeeae9df7f6/packages/%40aws-cdk/aws-route53/lib/util.ts#L26